blinkpy/blinkpy.py

```diff
@@ -220,8 +235,10 @@ def setup_camera_list(self):
                 if camera_network not in all_cameras:
                     all_cameras[camera_network] = []
                 for camera in network["cameras"]:
+                    # Keep only alphanumeric characters for name.
+                    camera_name = re.sub(r"\W+", "", camera["name"])
```

Comment:
I didn't want to have to add this, but Blink doesn't use special characters for camera names the local storage manifest.  I would have rather left it alone to be honest.

I had dashes and spaces in my camera names, but the local storage manifest returns the camera names without the dashes and spaces.  For example:  `Back-Door` or `Back Door` both become `BackDoor` in the manifest.  Underscores seem to be fine though.

Using a string as a key isn't really the best option in general.  If a user changes their camera's name, then all the old references would break anyway.  I changed the names of my cameras and the did a reload in HA from the Integrations page.  HA picked up the updated names, removed the old devices, and created new ones.  Of course I had to adjust any references to the old names in all my configuration.

I agree this is a breaking change, but appears to be necessary to support the way Blink returns camera names in the local storage manifest.  Please suggest an alternative if you see one.

IMO, a better approach would be to start using the unique camera IDs assigned by Blink as the camera keys.  This would require more current breaking changes, but would prevent future breaking changes.

Reply:
Removed.